diff -burN procps-3.1.12.orig/Makefile procps-3.1.12/Makefile
--- procps-3.1.12.orig/Makefile	2003-09-15 09:23:49.108280728 +0200
+++ procps-3.1.12/Makefile	2003-09-15 09:29:20.682873728 +0200
@@ -61,7 +61,8 @@
             uptime.c tload.c free.c w.c top.c vmstat.c watch.c skill.c \
             sysctl.c pgrep.c top.h pmap.c
 
-CURSES := -I/usr/include/ncurses -lncurses
+CURSESINC := -I/usr/include/ncurses
+CURSESLIB := -lncurses
 
 # Preprocessor flags.
 PKG_CPPFLAGS := -D_GNU_SOURCE -I proc
@@ -80,7 +81,8 @@
   -W -Wall -Wshadow -Wcast-align -Wredundant-decls \
   -Wbad-function-cast -Wcast-qual -Wwrite-strings -Waggregate-return \
   -Wstrict-prototypes -Wmissing-prototypes
-CFLAGS := -O2 -g3
+OPT := -O2
+CFLAGS := $(OPT) -g3
 ALL_CFLAGS := $(PKG_CFLAGS) $(CFLAGS)
 
 PKG_LDFLAGS := -Wl,-warn-common
@@ -163,16 +165,22 @@
 w.o:    w.c
 	$(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) $(W_SHOWFROM) -c $<
 
+top.o : top.c
+	$(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -c -o $@ $< $(CURSESINC)
+
+watch.o : watch.c
+	$(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -c -o $@ $< $(CURSESINC)
+
 ############ prog.o --> prog
 
 pmap w uptime tload free sysctl vmstat utmp pgrep skill: % : %.o $(LIBPROC)
 	$(CC) $(ALL_CFLAGS) $(ALL_LDFLAGS) -o $@ $^
 
 top:   % : %.o $(LIBPROC)
-	$(CC) $(ALL_CFLAGS) $(ALL_LDFLAGS) -o $@ $^ $(CURSES)
+	$(CC) $(ALL_CFLAGS) $(ALL_LDFLAGS) -o $@ $^ $(CURSESLIB)
 
 watch: % : %.o
-	$(CC) $(ALL_CFLAGS) $(ALL_LDFLAGS) -o $@ $^ $(CURSES)
+	$(CC) $(ALL_CFLAGS) $(ALL_LDFLAGS) -o $@ $^ $(CURSESLIB)
 
 ############ progX --> progY
 
diff -burN procps-3.1.12.orig/proc/module.mk procps-3.1.12/proc/module.mk
--- procps-3.1.12.orig/proc/module.mk	2003-09-15 09:23:49.104281336 +0200
+++ procps-3.1.12/proc/module.mk	2003-09-15 09:29:57.364297304 +0200
@@ -63,7 +63,7 @@
 $(lib)/$(SONAME) : proc/$(SONAME)
 	$(install) --mode a=rx --strip $< $@
 	cd $(lib) && $(ln_sf) $(SONAME) lib$(NAME).so
-	$(ldconfig)
+	-$(ldconfig)
 
 #$(usr/lib)/lib$(NAME).a : proc/lib$(NAME).a
 #	$(install) --mode a=r --strip $< $@
