diff -Nur procps-3.1.9/Makefile procps-3.1.9.new/Makefile
--- procps-3.1.9/Makefile	2003-06-06 03:39:56.000000000 +0200
+++ procps-3.1.9.new/Makefile	2003-06-06 03:41:49.000000000 +0200
@@ -156,13 +156,13 @@
 ############ prog.o --> prog
 
 pmap w uptime tload free sysctl vmstat utmp pgrep skill: % : %.o $(LIBPROC)
-	$(CC) $(LDFLAGS) -o $@ $^
+	$(CC) $(LDFLAGS) -o $@ $^ $(SELIB)
 
 top:   % : %.o $(LIBPROC)
-	$(CC) $(LDFLAGS) -o $@ $^ $(CURSESLIB)
+	$(CC) $(LDFLAGS) -o $@ $^ $(CURSESLIB) $(SELIB)
 
 watch: % : %.o
-	$(CC) $(LDFLAGS) -o $@ $^ $(CURSESLIB)
+	$(CC) $(LDFLAGS) -o $@ $^ $(CURSESLIB) $(SELIB)
 
 ############ progX --> progY
 
diff -Nur procps-3.1.9/proc/readproc.c procps-3.1.9.new/proc/readproc.c
--- procps-3.1.9/proc/readproc.c	2003-05-31 17:09:13.000000000 +0200
+++ procps-3.1.9.new/proc/readproc.c	2003-06-06 03:41:49.000000000 +0200
@@ -758,6 +758,8 @@
     else
 	PT = openproc(flags);
     va_end(ap);
+    if (!PT)
+      return 0;
     do {					/* read table: */
 	tab = xrealloc(tab, (n+1)*sizeof(proc_t*));/* realloc as we go, using */
 	tab[n] = readproc_direct(PT, NULL);     /* final null to terminate */
diff -Nur procps-3.1.9/ps/module.mk procps-3.1.9.new/ps/module.mk
--- procps-3.1.9/ps/module.mk	2003-03-30 22:04:07.000000000 +0200
+++ procps-3.1.9.new/ps/module.mk	2003-06-06 03:41:49.000000000 +0200
@@ -20,7 +20,7 @@
 TARFILES += $(PSSRC) $(addprefix ps/,$(PS_X))
 
 ps/ps: $(PSOBJ) $(LIBPROC)
-	$(CC) $(LDFLAGS) -o $@ $^
+	$(CC) $(LDFLAGS) -o $@ $^ $(SELIB)
 
 # This just adds the stacktrace code
 ps/debug: $(PSOBJ) stacktrace.o $(LIBPROC)
