--- procps-3.2.0/Makefile.orig	Mon Feb 23 18:21:20 2004
+++ procps-3.2.0/Makefile	Tue Feb 24 11:52:39 2004
@@ -69,7 +69,8 @@
 # plus the top-level Makefile to make it work stand-alone.
 _TARFILES := Makefile
 
-CURSES := -I/usr/include/ncurses -lncurses
+CURSESINC := -I/usr/include/ncurses
+CURSESLIB := -lncurses
 
 # Preprocessor flags.
 PKG_CPPFLAGS := -D_GNU_SOURCE -I proc
@@ -198,16 +199,25 @@
 w.o:    w.c
 	$(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) $(W_SHOWFROM) -c $<
 
+slabtop.o : slabtop.c
+	$(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -c -o $@ $< $(CURSESINC)
+
+top.o : top.c
+	$(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -c -o $@ $< $(CURSESINC)
+
+watch.o : watch.c
+	$(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -c -o $@ $< $(CURSESINC)
+
 ############ prog.o --> prog
 
 pmap w uptime tload free sysctl vmstat utmp pgrep skill: % : %.o $(LIBPROC)
 	$(CC) $(ALL_CFLAGS) $^ $(ALL_LDFLAGS) -o $@
 
 slabtop top: % : %.o $(LIBPROC)
-	$(CC) $(ALL_CFLAGS) $^ $(ALL_LDFLAGS) -o $@ $(CURSES)
+	$(CC) $(ALL_CFLAGS) $^ $(ALL_LDFLAGS) -o $@ $(CURSESLIB)
 
 watch: % : %.o
-	$(CC) $(ALL_CFLAGS) $^ $(ALL_LDFLAGS) -o $@ $(CURSES)
+	$(CC) $(ALL_CFLAGS) $^ $(ALL_LDFLAGS) -o $@ $(CURSESLIB)
 
 ############ progX --> progY
 
--- procps-3.1.12.orig/proc/module.mk	2003-09-15 09:23:49.104281336 +0200
+++ procps-3.1.12/proc/module.mk	2003-09-15 09:29:57.364297304 +0200
@@ -63,7 +63,7 @@
 $(lib)/$(SONAME) : proc/$(SONAME)
 	$(install) --mode a=rx --strip $< $@
 	cd $(lib) && $(ln_sf) $(SONAME) lib$(NAME).so
-	$(ldconfig)
+	-$(ldconfig)
 
 #$(usr/lib)/lib$(NAME).a : proc/lib$(NAME).a
 #	$(install) --mode a=r --strip $< $@
