diff -urN procps-3.1.0.org/Makefile procps-3.1.0/Makefile
--- procps-3.1.0.org/Makefile	Sat Nov  9 13:42:22 2002
+++ procps-3.1.0/Makefile	Sat Nov  9 13:58:35 2002
@@ -61,11 +61,13 @@
             uptime.c tload.c free.c w.c top.c vmstat.c watch.c skill.c \
             sysctl.c pgrep.c top.h pmap.c
 
-CURSES := -I/usr/include/ncurses -lncurses
+CURSESINC := -I/usr/include/ncurses
+CURSESLIB := -lncurses
 
 LDFLAGS := -Wl,-warn-common
 
-CFLAGS := -D_GNU_SOURCE -O2 -g3 -fno-common -ffast-math -I proc \
+OPT := -O2
+CFLAGS := $(OPT) -D_GNU_SOURCE -g3 -fno-common -ffast-math -I proc \
   -W -Wall -Wshadow -Wcast-align -Wredundant-decls \
   -Wbad-function-cast -Wcast-qual -Wwrite-strings -Waggregate-return \
 #  -Wpadded -Wunreachable-code -Wdisabled-optimization \
@@ -141,6 +143,12 @@
 %.o : %.c
 	$(CC) $(CFLAGS) -c -o $@ $^
 
+top.o : top.c
+	$(CC) $(CFLAGS) -c -o $@ $^ $(CURSESINC)
+
+watch.o : watch.c
+	$(CC) $(CFLAGS) -c -o $@ $^ $(CURSESINC)
+
 w.o:    w.c
 	$(CC) $(CFLAGS) $(W_SHOWFROM) -c $<
 
@@ -150,10 +158,10 @@
 	$(CC) $(LDFLAGS) -o $@ $^
 
 top:   % : %.o $(LIBPROC)
-	$(CC) $(LDFLAGS) -o $@ $^ $(CURSES)
+	$(CC) $(LDFLAGS) -o $@ $^ $(CURSESLIB)
 
 watch: % : %.o
-	$(CC) $(LDFLAGS) -o $@ $^ $(CURSES)
+	$(CC) $(LDFLAGS) -o $@ $^ $(CURSESLIB)
 
 ############ progX --> progY
 
diff -urN procps-3.1.0.org/proc/module.mk procps-3.1.0/proc/module.mk
--- procps-3.1.0.org/proc/module.mk	Sat Nov  9 13:42:22 2002
+++ procps-3.1.0/proc/module.mk	Sat Nov  9 14:02:00 2002
@@ -59,7 +59,7 @@
 $(lib)/$(SONAME) : proc/$(SONAME)
 	$(install) --mode a=rx --strip $< $@
 	cd $(lib) && $(ln_sf) $(SONAME) lib$(NAME).so
-	$(ldconfig)
+	-$(ldconfig)
 
 #$(usr/lib)/lib$(NAME).a : proc/lib$(NAME).a
 #	$(install) --mode a=r --strip $< $@
