--- ./proc/readproc.h.org	Wed Jul  2 01:27:26 2003
+++ ./proc/readproc.h	Mon Jul 28 18:35:01 2003
@@ -213,6 +213,8 @@
 #define PROC_FILLWCHAN  0x0080 /* look up WCHAN name */
 #define PROC_FILLARG    0x0100 /* alloc and fill in `cmdline' */
 
+// Backport from procps 2.0.7 - needed by pTools (what for?)
+#define PROC_FILLBUG	0x3f	/* No idea what we need */
 
 /* Obsolete, consider only processes with one of the passed: */
 #define PROC_PID     0x1000  /* process id numbers ( 0   terminated) */
