--- procps-981104/Makefile	Thu Oct 29 00:02:41 1998
+++ procps-981104.pld/Makefile	Wed Dec 30 13:02:30 1998
@@ -67,7 +67,7 @@
 # INSTALL PROGRAMS + DOCS
 install: $(patsubst %,install_%,$(BUILD) $(MAN1) $(MAN8))
 ifeq ($(SHARED),1)
-	install --owner 0 --group 0 --mode a=rx $(LIB_TGT) $(SHLIBDIR)
+	install --mode a=rx $(LIB_TGT) $(SHLIBDIR)
 endif
 
 # INSTALL LIBRARIES + HEADERS (OPTIONAL)
--- procps-981104/ps/Makefile	Tue Jul 14 10:23:18 1998
+++ procps-981104.pld/ps/Makefile	Wed Dec 30 12:47:05 1998
@@ -7,19 +7,21 @@
 #     -D_GNU_SOURCE -Wl,-warn-common -o ps ps.o proc/libproc.so.1.2
 
 #  Note that -O2 is needed for best warning support!
-FLAG=-O2 -s -W -Wall -Wstrict-prototypes -Wshadow -Wcast-align
+OPT= -O2
+FLAG= $(OPT) -s -W -Wall -Wstrict-prototypes -Wshadow -Wcast-align
 #FLAG=-O0 -g3 -DDEBUG
 CC=gcc $(FLAG) -D_GNU_SOURCE
 #CC=gcc -pg -O2 -g3 -D_GNU_SOURCE
 
 # These things are needed to make some stuff work... ?
+DESTDIR    =
 SHARED     =  1# build/install both a static and ELF shared library
 LDFLAGS    =  -Wl,-warn-common #-s      recommended for ELF systems
 BFD_CAPABLE = -DBFD_CAPABLE
 AOUT_CAPABLE = #-DAOUT_CAPABLE 
 ELF_CAPABLE = #-DELF_CAPABLE
 LIBBFD = -lbfd -liberty
-MAN1DIR = /usr/man/man1
+MAN1DIR = $(DESTDIR)/usr/man/man1
 
 all: ps
 
--- procps-2.0/ps/Makefile.old	Sat Mar 13 09:25:29 1999
+++ procps-2.0/ps/Makefile	Sat Mar 13 09:32:49 1999
@@ -52,8 +52,8 @@
 
 
 install: ps
-	-install --owner 0 --group 0 --mode a=rx --strip ps ${DESTDIR}/bin/ps
-	-install --owner 0 --group 0 --mode a=r ps.1 ${DESTDIR}$(MAN1DIR)/ps.1
+	-install --mode a=rx --strip ps ${DESTDIR}/bin/ps
+	-install --mode a=r ps.1 ${DESTDIR}$(MAN1DIR)/ps.1
 	rm -f ${DESTDIR}/var/catman/cat1/ps.1.gz ${DESTDIR}/var/man/cat1/ps.1.gz
 
 clean:
