diff -urN procps-3.2.4.orig/ps/parser.c procps-3.2.4/ps/parser.c
--- procps-3.2.4.orig/ps/parser.c	2004-11-05 10:12:38.993641616 +0100
+++ procps-3.2.4/ps/parser.c	2004-11-05 10:18:54.991481256 +0100
@@ -226,7 +226,7 @@
     // This list of options is ONLY for those defined by the
     // "IEEE Std 1003.1, 2004 Edition", "ISO/IEC 9945:2003",
     // or "Version 2 of the Single Unix Specification".
-    if(!strchr("aAdefgGlnoptuU", *flagptr)) not_pure_unix = 1;
+    if(!strchr("aAdefgGlnoptuUzZ", *flagptr)) not_pure_unix = 1;
 
     switch(*flagptr){
     case 'A':
@@ -331,12 +331,19 @@
       exclusive("-V");
       display_version();
       exit(0);
+#ifdef __linux__
+    case 'Z':     /* full Mandatory Access Control level info */
+      trace("-Z shows full MAC info\n");
+      format_modifiers |= FF_Fc;
+      break;
+#else
     // This must be verified against SVR4-MP. (UnixWare or Powermax)
     // Leave it undocumented until that problem is solved.
     case 'Z':     /* full Mandatory Access Control level info */
       trace("-Z shows full MAC info\n");
       format_modifiers |= FM_M;
       break;
+#endif
     case 'a':
       trace("-a select all with a tty, but omit session leaders.\n");
       simple_select |= SS_U_a;
@@ -467,13 +474,11 @@
       trace("-y Print lnone info in UID/USER column or do Sun -l hack.\n");
       format_modifiers |= FM_y;
       break;
-#if 0
     // This must be verified against SVR4-MP (UnixWare or Powermax)
     case 'z':     /* alias of Mandatory Access Control level info */
-      trace("-z shows aliased MAC info\n");
-      format_modifiers |= FM_M;
+      trace("-z shows full MAC info\n");
+      format_modifiers |= FF_Fc;
       break;
-#endif
     case '-':
       return "Embedded '-' among SysV options makes no sense.";
       break;
@@ -591,10 +596,17 @@
       trace("X Old Linux i386 register format\n");
       format_flags |= FF_LX;
       break;
+#ifdef __linux__
+    case 'Z':
+      trace("Z SElinux Security Context format\n");
+      format_flags |= FF_Fc;
+      break;
+#else
     case 'Z':  /* FreeBSD does MAC like SGI's Irix does it */
       trace("Z Print security label for Mandatory Access Control.\n");
       format_modifiers |= FM_M;
       break;
+#endif
     case 'a':
       trace("a Select all w/tty, including other users\n");
       simple_select |= SS_B_a;
diff -urN procps-3.2.4.orig/ps/ps.1 procps-3.2.4/ps/ps.1
--- procps-3.2.4.orig/ps/ps.1	2004-11-05 10:12:38.996641160 +0100
+++ procps-3.2.4/ps/ps.1	2004-11-05 10:14:16.035888952 +0100
@@ -491,6 +491,10 @@
 display virtual memory format
 
 .TP
+.B \-Z
+display security context format (NSA SELinux, etc.)
+
+.TP
 .B \-y
 Do not show flags; show rss in place of addr.
 This option can only be used with \fB\-l\fR.
