diff -ruN procps-2.0.3.orig/w.c procps-2.0.3/w.c
--- procps-2.0.3.orig/w.c	Fri Jul  2 22:17:04 1999
+++ procps-2.0.3/w.c	Sun Sep 19 11:35:55 1999
@@ -170,8 +170,10 @@
 	 * is a desire a cmdline flag could be added to optionally show it with a
 	 * prefix of (stale) in front of cmd or something like that.
 	 */
+	/*
 	if (!ut_pid_found)
 		return;
+	*/
 
 	strncpy(uname, u->ut_user, USERSZ);		/* force NUL term for printf */
 	if (formtype) {
