--- procps-2.0.6/proc/Makefile.sig	Tue Nov  2 09:02:16 1999
+++ procps-2.0.6/proc/Makefile	Tue Jan 11 17:42:34 2000
@@ -101,13 +101,3 @@ ifdef MINORVERSION
 else
 	$(strip $(CC) $(CFLAGS) -DVERSION=\"$(VERSION)\" -DSUBVERSION=\"$(SUBVERSION)\" -c version.c)
 endif
-
-signals.o : signames.h
-
-signames.h ../proc/signames.h : /usr/include/signal.h
-	$(CPP) -dM /usr/include/signal.h |	\
-	grep -v SIGSTKSZ | \
-	tr -s '\t ' '  ' | sort -n +2 | sed		\
-	's:#define SIG\([A-Z]\+[0-9]*\) \([0-9]\+\) *\(\|/\*.*\)$$:{\
-\2,"\1" },:p;d'  > signames.h
-
--- procps-2.0.6/proc/signals.c.sig	Fri Mar 12 02:34:44 1999
+++ procps-2.0.6/proc/signals.c	Tue Jan 11 17:41:49 2000
@@ -23,8 +23,143 @@ typedef struct {
 
 
 static SIGNAME signals[] = {
-#include "signames.h" /* should be in same dir as this file */
-  { 0,NULL }};
+#ifdef SIGHUP
+  { SIGHUP, "HUP" },
+#endif
+#ifdef SIGINT
+  { SIGINT, "INT" },
+#endif
+#ifdef SIGQUIT
+  { SIGQUIT, "QUIT" },
+#endif
+#ifdef SIGILL
+  { SIGILL, "ILL" },
+#endif
+#ifdef SIGTRAP
+  { SIGTRAP, "TRAP" },
+#endif
+#ifdef SIGABRT
+  { SIGABRT, "ABRT" },
+#endif
+#ifdef SIGIOT
+  { SIGIOT, "IOT" },
+#endif
+#ifdef SIGBUS
+  { SIGBUS, "BUS" },
+#endif
+#ifdef SIGFPE
+  { SIGFPE, "FPE" },
+#endif
+#ifdef SIGKILL
+  { SIGKILL, "KILL" },
+#endif
+#ifdef SIGUSR1
+  { SIGUSR1, "USR1" },
+#endif
+#ifdef SIGSEGV
+  { SIGSEGV, "SEGV" },
+#endif
+#ifdef SIGUSR2
+  { SIGUSR2, "USR2" },
+#endif
+#ifdef SIGPIPE
+  { SIGPIPE, "PIPE" },
+#endif
+#ifdef SIGALRM
+  { SIGALRM, "ALRM" },
+#endif
+#ifdef SIGTERM
+  { SIGTERM, "TERM" },
+#endif
+#ifdef SIGSTKFLT
+  { SIGSTKFLT, "STKFLT" },
+#endif
+#ifdef SIGCHLD
+  { SIGCHLD, "CHLD" },
+#endif
+#ifdef SIGCONT
+  { SIGCONT, "CONT" },
+#endif
+#ifdef SIGSTOP
+  { SIGSTOP, "STOP" },
+#endif
+#ifdef SIGTSTP
+  { SIGTSTP, "TSTP" },
+#endif
+#ifdef SIGTTIN
+  { SIGTTIN, "TTIN" },
+#endif
+#ifdef SIGTOU
+  { SIGTOU, "TOU" },
+#endif
+#ifdef SIGURG
+  { SIGURG, "URG" },
+#endif
+#ifdef SIGXCPU
+  { SIGXCPU, "XCPU" },
+#endif
+#ifdef SIGXFSZ
+  { SIGXFSZ, "XFSZ" },
+#endif
+#ifdef SIGVTALRM
+  { SIGVTALRM, "VTALRM" },
+#endif
+#ifdef SIGPROF
+  { SIGPROF, "PROF" },
+#endif
+#ifdef SIGWINCH
+  { SIGWINCH, "WINCH" },
+#endif
+#ifdef SIGIO
+  { SIGIO, "IO" },
+#endif
+#ifdef SIGPWR
+  { SIGPWR, "PWR" },
+#endif
+#ifdef SIGUNUSED
+  { SIGUNUSED, "UNUSED" },
+#endif
+#ifdef SIGEMT
+  { SIGEMT, "EMT" },
+#endif
+#ifdef SIGSYS
+  { SIGSYS, "SYS" },
+#endif
+#ifdef SIGWIND
+  { SIGWIND, "WIND" },
+#endif
+#ifdef SIGPHONE
+  { SIGPHONE, "PHONE" },
+#endif
+#ifdef SIGLOST
+  { SIGLOST, "LOST" },
+#endif
+#ifdef SIGWAITING
+  { SIGWAITING, "WAITING" },
+#endif
+#ifdef SIGLWP
+  { SIGLWP, "LWP" },
+#endif
+#ifdef SIGDANGER
+  { SIGDANGER, "DANGER" },
+#endif
+#ifdef SIGGRANT
+  { SIGGRANT, "GRANT" },
+#endif
+#ifdef SIGRETRACT
+  { SIGRETRACT, "RETRACT" },
+#endif
+#ifdef SIGMSG
+  { SIGMSG, "MSG" },
+#endif
+#ifdef SIGSOUND
+  { SIGSOUND, "SOUND" },
+#endif
+#ifdef SIGSAK
+  { SIGSAK, "SAK" },
+#endif
+  { 0,NULL }
+};
 
 
 void list_signals(void)
